src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -81,22 +78,27 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
               IndexLogEntry.schemaString(schema),
               numBuckets)),
           Content(path.toString, Seq()),
-          Source(SparkPlan(sourcePlanProperties), Seq(Hdfs(sourceDataProperties))),
+          Source(
+            SparkPlan(sourcePlanProperties), Seq(Hdfs(sourceDataProperties)), relations),
           Map())
 
       case None => throw HyperspaceException("Invalid plan for creating an index.")
     }
   }
 
-  protected def sourceFiles(df: DataFrame): Seq[String] =
+  protected def sourceRelations(df: DataFrame): Seq[RelationMetadataEntry] =
     df.queryExecution.optimizedPlan.collect {
       case LogicalRelation(
-          HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
+          HadoopFsRelation(location: PartitioningAwareFileIndex, _, dataSchema, _, fileFormat, _),
           _,
           _,
           _) =>
-        location.allFiles.map(_.getPath.toString)
-    }.flatten
+        RelationMetadataEntry(
+          location.rootPaths.map(_.toString),
+          location.allFiles.map(_.getPath.toString),
+          dataSchema.json,
+          fileFormat.toString.toLowerCase(Locale.ROOT))
```

Comment:
`fileFormat.shortName` is better since it's an abstract method? (also used in `DataSource.lookupDataSource`)

Reply:
Yes but seems it's not accessible with FileFormat..?

Reply:
Can you try:
```scala
fileFormat match {
  case d: DataSourceRegister => d.shortName
  case _ => throw HyperspaceException("blah")
}
```
?

Reply:
It woks well thanks :)